src/test/App.test.tsx

```diff
@@ -0,0 +1,56 @@
+import '@testing-library/jest-dom'
+import { render, screen, waitFor } from '@testing-library/react'
+import { BigNumber } from 'ethers'
+import React from 'react'
+import { App } from './App'
+
+const MAX_BLOCK_AGE = 600_000 // 10 minutes
+
+describe('Use multicall in test application', () => {
+  beforeAll(() => {
+    if (!process.env.INFURA_PROJECT_ID) {
+      throw new Error('Test requires INFURA_PROJECT_ID env var to be set')
+    }
+  })
+
+  it('Renders correctly initially', () => {
+    render(<App />)
+    const h1 = screen.getByText('Hello Multicall') // H1 in Home.tsx
+    expect(h1).toBeTruthy()
+    const missing = screen.queryByText('Does Not Exist')
+    expect(missing).toBeFalsy()
+  })
+
+  it('Performs a single contract multicall query', async () => {
+    render(<App />)
+    // Check that block timestamp is correctly retrieved from block
+    const timestamp1 = await waitFor(() => screen.getByTestId('blockTimestamp'), { timeout: 15_000 /* 15 seconds */ })
+    expect(timestamp1 && timestamp1?.textContent).toBeTruthy()
+    const value1 = parseInt(timestamp1.textContent!, 10) * 1000
+    const now1 = Date.now()
+    expect(now1 - value1).toBeLessThan(MAX_BLOCK_AGE)
+
+    // Wait for an updated block timestamp
+    await sleep(10_000 /* 10 seconds */)
```

Comment:
this is gonna be pretty flaky isn't it? block times are 12 seconds on mainnet

Reply:
Worked well so far but I'll increase the times for reliability